src/Logger/Adapter.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Utopia\Logger;
+
+use Exception;
+
+abstract class Adapter
+{
+    /**
+     * Get unique name of an adapter
+     *
+     * @return string
+     */
+    abstract public static function getName(): string;
+
+
+    /**
+     * Push log to external provider
+     *
+     * @param Log $log
+     * @return int
+     * @throws Exception
+     */
+    abstract public function pushLog(Log $log): int;
```

Comment:
I think it will make more sense to call `pushLog` just `push`, taking into account we're already in the logger namespace.